src/script/finale.ts

```diff
@@ -15,6 +22,22 @@ document.addEventListener("DOMContentLoaded", () => {
 }, false);
 
 let originPath = "";
+
+function LoadDOM(query : string) {
+    const viewObj = new ViewFactory();
+    const htmlDOM = viewObj.generate(Main.viModel);
+    if (htmlDOM) {
+        const url = document.location.search + document.location.hash;
+        HistoryManager.UpdateHistory(url, Main.viModel);
```

Comment:
Can you please add a TODO here? I have a feeling this can probably be removed from here. Updating history inside LoadFile(), where the sessionStorage is being set should probably suffice. I'm not sure, so we can make that fix later (may or may not be in this same PR).

Reply:
Done